### PR TITLE
shared/tinyusb: Create for reuse, rp2 only for now.

### PR DESCRIPTION
**Pull request by Molorius**

### CODE REVIEW

File: ports/rp2/usbd.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#include "usbd.h"
+#include "string.h"
+#include "pico/unique_id.h"
+
+int usbd_serialnumber(uint8_t *buf) {
+    pico_unique_board_id_t id;
+    const int len = 8;
+
+    pico_get_unique_board_id(&id);
+    memcpy(buf, id.id, len);
```
Comment (projectgus) [inline ports/rp2/usbd.c:36]: suggest using `sizeof(id.id)` here and for the return value

(there is also a constant `PICO_UNIQUE_BOARD_ID_SIZE_BYTES` which would be fine, but `sizeof(id.id)` seems like the best choice to me as that's the actual field you're copying from.)

**Review by projectgus — CHANGES_REQUESTED** [review]
Generally this looks good to me, just a couple of minor things! Sorry to ask you to leave most of your work behind for now, but this is a really good foundation.

There is also a file `shared/runtime/tinyusb_helpers.c` which contains the implementation for `MICROPY_HW_USB_CDC_1200BPS_TOUCH`. It might make sense to rename this to `shared/tinyusb/usbd_cdc_common.c` or something like that, I'm not sure. But feel free to leave that for now.

**Comment by projectgus** [comment]
@Molorius I used your commit as the basis for a branch to implement USB device drivers in Python (see linked PR). I also have a separate refactoring commit in that PR - if it's OK with you then I'll submit a new USB refactoring PR with your commit plus that one.

**Comment by Molorius** [comment]
Hello @projectgus yes it is okay with me. Thanks for asking.

**Comment by dpgeorge** [comment]
@projectgus how shall we proceed here?  Do you want to submit a separate PR (separate to this and #9497) to do the refactoring?  Otherwise I see it's already part of #9497 and can be reviewed with that set of commits, although that's not my preference because it's easier to review and merge smaller chunks of work.

**Comment by projectgus** [comment]
@dpgeorge Yes, my plan has been to submit a separate refactor PR. I wanted to wait until hearing back from @Molorius (thanks!), and also until I thought the refactor changes were finalised and I wasn't going to do much more tweaking.

I'm probably are at this point now, so I'll submit a refactoring PR today.

**Comment by dpgeorge** [comment]
This was merged in eed4eb2645aa3975ef822f4b34110730811690b7 as part of #9816.

Thanks @Molorius for initiating this refactor!

**Pull request closed without merging**